src/libarena/lib.rs

```diff
@@ -132,7 +134,9 @@ impl<T> TypedArena<T> {
 
         unsafe {
             if mem::size_of::<T>() == 0 {
-                self.ptr.set(intrinsics::arith_offset(self.ptr.get() as *mut u8, 1) as *mut T);
+                self.ptr
+                    .set(intrinsics::arith_offset(self.ptr.get() as *mut u8, 1)
+                        as *mut T);
```

Comment:
this seems wrong. identical construct below puts `intrinsics` -- `as *mut T` on one line

Reply:
Why do we use `intrinsics::arith_offset(x, 1)` instead of `x.wrapping_offset(1)` anyway?

Reply:
@mrhota Should I submit an issue to the Rustfmt repo?

Reply:
If we're comparing this to (new) line 369, the difference is because line 369 is one level outdented from this line. That's the reason for the difference.

Still seems like bad formatting, though.